KalturaWowzaServer/src/main/java/com/kaltura/media_server/modules/AuthenticationModule.java

```diff
@@ -72,25 +72,48 @@ private KalturaLiveEntry onClientConnect(WMSProperties properties, HashMap<Strin
 
         int partnerId = Integer.parseInt(requestParams.get(Constants.REQUEST_PROPERTY_PARTNER_ID));
         String entryId = requestParams.get(Constants.REQUEST_PROPERTY_ENTRY_ID);
-        String token = requestParams.get(Constants.REQUEST_PROPERTY_TOKEN);
-
         KalturaEntryServerNodeType serverIndex = KalturaEntryServerNodeType.get(requestParams.get(Constants.REQUEST_PROPERTY_SERVER_INDEX));
-        KalturaLiveEntry liveEntry = KalturaAPI.getKalturaAPI().authenticate(entryId, partnerId, token, serverIndex);
-        KalturaEntryDataPersistence.setProperty(entryId, Constants.CLIENT_PROPERTY_KALTURA_LIVE_ENTRY, liveEntry);
-        synchronized(properties) {
-            properties.setProperty(Constants.CLIENT_PROPERTY_SERVER_INDEX, requestParams.get(Constants.REQUEST_PROPERTY_SERVER_INDEX));
-            properties.setProperty(Constants.KALTURA_LIVE_ENTRY_ID, liveEntry.id);
+        Object authenticationLock = KalturaEntryDataPersistence.setLock(entryId);
+        synchronized (authenticationLock) {
+            try {
+                if ((boolean)KalturaEntryDataPersistence.getPropertyByEntry(entryId, Constants.KALTURA_ENTRY_AUTHENTICATION_ERROR_FLAG)) {
+                    throw new Exception("(" + entryId + ") Authentication failed for entry!");
+                }
+                String token = requestParams.get(Constants.REQUEST_PROPERTY_TOKEN);
```

Comment:
suggesting to do all requestParams.get(...) outside of the synchronized block. Preferable that the block will contains minimum code. It's ok if ingest will get request params but won't do anything with them because other ingest already did authentication.